src/main/java/org/jabref/gui/entryeditor/AiChatTab.java

```diff
@@ -1,271 +1,81 @@
-                    .onSuccess(res -> handleFocus())
-                    .onFailure(this::showErrorWhileIngesting)
-                    .executeWith(taskExecutor);
-        }
-    }
-
-    private void showErrorWhileIngesting(Exception e) {
-        LOGGER.error("Got an error while generating embeddings for entry {}", currentEntry.getCitationKey(), e);
-
-        setContent(ErrorStateComponent.withTextArea(Localization.lang("Unable to chat"), Localization.lang("Got error while processing the file:"), e.getMessage()));
-
-        entriesUnderIngestion.remove(currentEntry);
-
-        currentEntry.getFiles().stream().map(LinkedFile::getLink).forEach(link -> aiService.getEmbeddingsManager().removeDocument(link));
-    }
-
-    private void bindToCorrectEntry(BibEntry entry) {
-        assert entry.getCitationKey().isPresent();
-
-        AiChatHistory aiChatHistory = getAiChatHistory(aiService, entry, bibDatabaseContext);
-
-        AiChatLogic aiChatLogic = AiChatLogic.forBibEntry(aiService, aiChatHistory, entry);
-
-        Node content = new AiChatComponent(aiService.getPreferences(), aiChatLogic, entry.getCitationKey().get(), dialogService, taskExecutor);
-
-        setContent(content);
-    }
-
-    private static AiChatHistory getAiChatHistory(AiService aiService, BibEntry entry, BibDatabaseContext bibDatabaseContext) {
-        Optional<Path> databasePath = bibDatabaseContext.getDatabasePath();
-
-        if (databasePath.isEmpty() || entry.getCitationKey().isEmpty()) {
-            LOGGER.warn("AI chat is constructed, but the database path is empty. Cannot store chat history");
-            return new InMemoryAiChatHistory();
-        } else if (entry.getCitationKey().isEmpty()) {
-            LOGGER.warn("AI chat is constructed, but the entry citation key is empty. Cannot store chat history");
-            return new InMemoryAiChatHistory();
-        } else {
-            return aiService.getChatHistoryManager().getChatHistory(databasePath.get(), entry.getCitationKey().get());
+        if (currentEntry != null) {
+            aiService.getChatHistoryService().closeChatHistoryForEntry(currentEntry);
         }
-    }
-
-    @Subscribe
-    public void listen(FullyIngestedDocumentsTracker.DocumentIngestedEvent event) {
-        UiTaskExecutor.runInJavaFXThread(AiChatTab.this::handleFocus);
-    }
-
-    @Subscribe
-    public void listen(JabRefEmbeddingModel.EmbeddingModelBuiltEvent event) {
-        UiTaskExecutor.runInJavaFXThread(AiChatTab.this::handleFocus);
-    }
 
-    @Subscribe
-    public void listen(JabRefEmbeddingModel.EmbeddingModelBuildingErrorEvent event) {
-        UiTaskExecutor.runInJavaFXThread(AiChatTab.this::handleFocus);
+        StringProperty chatName = new SimpleStringProperty("entry " + entry.getCitationKey().orElse("<no citation key>"));
+        entry.getCiteKeyBinding().addListener((observable, oldValue, newValue) -> chatName.setValue("entry " + newValue));
```

Comment:
This needs to be Localized.

(no localization for "no citatoin key" is OK, because this never happens. Add a code comemnt thoguh). 

Do you need a check for an empty key at the `addListener`? Mabe, then we need "no citation key" localized?

Reply:
Oops, it should be localized because it's reused in the title of a group chat window.

Initially it's only used for logging purposes

Reply:
Oops, this is a little bit complex case --> add more comments.

This name property is used for two purposes:
1. Logging - no localization needed.
2. Window title for group chats - localization needed.

So, in this place we may not localize it.

But for the group window we should.

I add comments in those places + field in AiChatLogic